src/components/EditRow.js

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from 'react'
+import Select from 'react-select'
+import 'react-select/dist/react-select.css'
+import '../css/EditRow.css'
+
+class EditRow extends Component {
+  constructor (props) {
+    super(props)
+    this.optionsFromDataTypes = this.optionsFromDataTypes.bind(this)
+    this.handleChange = this.handleChange.bind(this)
+    this.handleSelectChange = this.handleSelectChange.bind(this)
+    this.handleSave = this.handleSave.bind(this)
+  }
+
+  optionsFromDataTypes () {
+    let options = []
+    Object.keys(this.props.dataTypes).map((dataType, idx) => {
+      return options.push({value: dataType, label: dataType})
+    })
+    return options
+  }
+
+  handleChange (event) {
+    const target = event.target
+    const value = target.type === 'checkbox' ? target.checked : target.value
+    let newState = Object.assign({}, this.props.details)
+    newState[target.name] = value
+    this.props.actions.updateRow(newState)
+  }
+
+  handleSave (event) {
+    event.preventDefault()
+    // const newRow = Object.assign({}, this.props.details, {edit: false})
+    // this.props.actions.updateRow(newRow)
+    this.props.actions.toggleEditRow(this.props.details.tableID, this.props.details.id)
+  }
+
+  handleSelectChange (selectedOption) {
+    if (selectedOption === null) { return }
+    let newState = Object.assign({}, this.props.details)
+    newState.dataType = selectedOption.value
+    this.props.actions.updateRow(newState)
+  }
+
+  render () {
+    const details = this.props.details
+    return (
+      <div className='EditRow' style={{backgroundColor: `${this.props.dataTypes[details.dataType].color}`}}>
```

Comment:
one thing you might consider to avoid inline styling is to have a global colors css file where the class name results in the background color. 

check out https://www.npmjs.com/package/classnames for using multiple class names